script/tool/lib/src/drive_examples_command.dart

```diff
@@ -43,213 +48,252 @@ class DriveExamplesCommand extends PluginCommand {
 
   @override
   final String description = 'Runs driver tests for plugin example apps.\n\n'
-      'For each *_test.dart in test_driver/ it drives an application with a '
-      'corresponding name in the test/ or test_driver/ directories.\n\n'
-      'For example, test_driver/app_test.dart would match test/app.dart.\n\n'
-      'This command requires "flutter" to be in your path.\n\n'
-      'If a file with a corresponding name cannot be found, this driver file'
-      'will be used to drive the tests that match '
-      'integration_test/*_test.dart.';
+      'For each *_test.dart in test_driver/ it drives an application with the '
+      'either the corresponding test in test_driver (for example, '
+      'test_driver/app_test.dart would match test_driver/app.dart), or the '
+      '*_test.dart files in integration_test/.\n\n'
+      'This command requires "flutter" to be in your path.';
+
+  Map<String, List<String>> _targetDeviceFlags = const <String, List<String>>{};
 
   @override
-  Future<void> run() async {
-    final List<String> failingTests = <String>[];
-    final List<String> pluginsWithoutTests = <String>[];
-    final bool isLinux = getBoolArg(kPlatformLinux);
-    final bool isMacos = getBoolArg(kPlatformMacos);
-    final bool isWeb = getBoolArg(kPlatformWeb);
-    final bool isWindows = getBoolArg(kPlatformWindows);
-    await for (final Directory plugin in getPlugins()) {
-      final String pluginName = plugin.basename;
-      if (pluginName.endsWith('_platform_interface') &&
-          !plugin.childDirectory('example').existsSync()) {
-        // Platform interface packages generally aren't intended to have
-        // examples, and don't need integration tests, so silently skip them
-        // unless for some reason there is an example directory.
-        continue;
+  Future<void> initializeRun() async {
+    final List<String> platformSwitches = <String>[
+      kPlatformAndroid,
+      kPlatformIos,
+      kPlatformLinux,
+      kPlatformMacos,
+      kPlatformWeb,
+      kPlatformWindows,
+    ];
+    final int platformCount = platformSwitches
+        .where((String platform) => getBoolArg(platform))
+        .length;
+    // The flutter tool currently doesn't accept multiple device arguments:
+    // https://github.com/flutter/flutter/issues/35733
+    // If that is implemented, this check can be relaxed.
+    if (platformCount != 1) {
+      printError(
+          'Exactly one of ${platformSwitches.map((String platform) => '--$platform').join(', ')} '
+          'must be specified.');
+      throw ToolExit(_exitNoPlatformFlags);
+    }
+
+    String? androidDevice;
+    if (getBoolArg(kPlatformAndroid)) {
+      final List<String> devices = await _getDevicesForPlatform('android');
+      if (devices.isEmpty) {
+        printError('No Android devices available');
+        throw ToolExit(_exitNoAvailableDevice);
+      }
+      androidDevice = devices.first;
```

Comment:
Should we warn if there are more than one device?

Reply:
I thought about adding more control over devices via flags, but I figured that absent anyone actually wanting that functionality I'd just do the simplest thing for now, and we can expand later (e.g., adding device choice flags to the command itself) if needed.